x-pack/docs/en/security/authentication/security-domain.asciidoc

```diff
@@ -41,24 +41,52 @@ users with the same username, it's reasonable to treat these users as the
 same user from a resource ownership perspective.
 
 Security domains make resource sharing across realms possible by grouping those
-realms under the same domain. {es} always enforces the privileges that are 
-associated with the currently authenticated user, which remains true with 
+realms under the same domain. {es} always enforces the privileges that are
+associated with the currently authenticated user, which remains true with
 security domains. Security domains don't bypass
-<<authorization,user authorization>> when resource sharing requires them. For 
-example, a user requires the `manage_own_api_key` privilege to manage their own 
-API keys. If that user doesn't have this privilege when authenticating with one 
-realm, they won't be able to manage API keys while authenticating with another 
+<<authorization,user authorization>> when resource sharing requires them. For
+example, a user requires the `manage_own_api_key` privilege to manage their own
+API keys. If that user doesn't have this privilege when authenticating with one
+realm, they won't be able to manage API keys while authenticating with another
 realm.
 
 [[security-domain-realm-roles]]
 ===== Managing roles across realms
 {es} provides multiple ways to consistently apply roles across realms. For example, you can use
-<<configuring-authorization-delegation,authorization delegation>> to 
+<<configuring-authorization-delegation,authorization delegation>> to
 ensure that a user is assigned the same roles from multiple realms. You can also
 manually configure multiple realms that are backed by the same directory service.
 Though it's possible to configure different <<roles,roles>> for the same user
 when authenticating with different realms, it is _not_ recommended.
 
+[[security-domain-warning]]]]
+[IMPORTANT]
+====
+Security domain is an advanced feature and must be used with care. It can lead
+to unexpected behaviours if incorrectly configured or misused.
+Here are few things watch out when working with security domain:
+
+* Security domains must be configured consistently across all nodes. Since changing
+security domain requires node restart, this means they may not be fully functional
+during rolling restart.
```

Comment:
Can we be more specific here?
As it reads, it says "be careful" and "rolling restarts might not work well", but doesn't tell the admin how to avoid potential problems. Do we just expect them not to do any changes? Or do full cluster restarts?

It would be nice to tell them something that they can actually respond to (if that's possible).

Reply:
I expanded a bit (dos and donts) on this.